agent/api/transitiondependency.go

```diff
@@ -19,3 +39,35 @@ type ContainerDependency struct {
 	// resource satisfies the dependency
 	DependentStatus ContainerStatus `json:"DependentStatus,omitempty"`
 }
+
+// TransitionDependenciesMap is a map of the dependent container status to other
+// dependencies that must be satisfied.
+type TransitionDependenciesMap map[ContainerStatus]TransitionDependencySet
+
+// UnmarshalJSON decodes the TransitionDependencySet tag in the JSON encoded string
+// into the TransitionDependenciesMap object
+func (td *TransitionDependenciesMap) UnmarshalJSON(b []byte) error {
+	depMap := make(map[ContainerStatus]TransitionDependencySet)
+	if err := json.Unmarshal(b, &depMap); err != nil {
+		seelog.Debugf("Unmarshal 'TransitionDependencySet': %s, not a map: %v", string(b), err)
```

Comment:
I'm not sure if we really need a `Debugf` statement here.

Reply:
It just helps know when we are dealing with old state files.